src/map/achievement.c

```diff
@@ -0,0 +1,1693 @@
+
+	achievement->validate_type(sd, ACH_PARTY_CREATE, &criteria, 1, true);
+}
+
+/**
+ * Validates marriage type objectives.
+ * @type ACH_MARRY
+ * @param sd       pointer to session data.
+ * @see achievement_validate_type()
+ */
+void achievement_validate_marry(struct map_session_data *sd)
+{
+	struct achievement_objective criteria = { 0 };
+
+	nullpo_retv(sd);
+
+	achievement->validate_type(sd, ACH_MARRY, &criteria, 1, true);
+}
+
+/**
+ * Validates adoption type objectives.
+ * @types ACH_ADOPT_PARENT
+ *        ACH_ADOPT_BABY
+ * @param sd       pointer to session data.
+ * @param parent   (type) boolean value to indicate if parent (true) or baby (false).
+ * @see achievement_validate_type()
+ */
+void achievement_validate_adopt(struct map_session_data *sd, bool parent)
+{
+	struct achievement_objective criteria = { 0 };
+
+	nullpo_retv(sd);
+
+	if (parent)
+		achievement->validate_type(sd, ACH_ADOPT_PARENT, &criteria, 1, true);
+	else
+		achievement->validate_type(sd, ACH_ADOPT_BABY, &criteria, 1, true);
+}
+
+/**
+ * Validates zeny type objectives.
+ * @types ACH_ZENY_HOLD
+ *        ACH_ZENY_GET_ONCE
+ *        ACH_ZENY_GET_TOTAL
+ *        ACH_ZENY_SPEND_ONCE
+ *        ACH_ZENY_SPEND_TOTAL
+ * @param sd      pointer to session data.
+ * @param amount  (goal) amount of zeny earned or spent.
+ * @see achievement_validate_type()
+ */
+void achievement_validate_zeny(struct map_session_data *sd, int amount)
+{
+	struct achievement_objective criteria = { 0 };
+
+	nullpo_retv(sd);
+	Assert_retv(amount != 0);
+
+	if (amount > 0) {
+		achievement->validate_type(sd, ACH_ZENY_HOLD, &criteria, sd->status.zeny, false);
+		achievement->validate_type(sd, ACH_ZENY_GET_ONCE, &criteria, amount, false);
+		achievement->validate_type(sd, ACH_ZENY_GET_TOTAL, &criteria, amount, true);
+	}  else {
+		achievement->validate_type(sd, ACH_ZENY_SPEND_ONCE, &criteria, amount, false);
+		achievement->validate_type(sd, ACH_ZENY_SPEND_TOTAL, &criteria, amount, true);
+	}
+}
+
+/**
+ * Validates equipment refinement type objectives.
+ * @types ACH_EQUIP_REFINE_SUCCESS
+ *        ACH_EQUIP_REFINE_FAILURE
+ *        ACH_EQUIP_REFINE_SUCCESS_TOTAL
+ *        ACH_EQUIP_REFINE_FAILURE_TOTAL
+ *        ACH_EQUIP_REFINE_SUCCESS_WLV
+ *        ACH_EQUIP_REFINE_FAILURE_WLV
+ *        ACH_EQUIP_REFINE_SUCCESS_ID
+ *        ACH_EQUIP_REFINE_FAILURE_ID
+ * @param sd       pointer to session data.
+ * @param idx      Inventory index of the item.
+ * @param success  (type) boolean switch for failure / success.
+ * @see achievement_validate_type()
+ */
+void achievement_validate_refine(struct map_session_data *sd, const unsigned int idx, bool success)
+{
+	struct achievement_objective criteria = { 0 };
+	struct item_data *id = itemdb->exists(sd->status.inventory[idx].nameid);
+
+	nullpo_retv(sd);
+	Assert_retv(idx < MAX_INVENTORY);
+	Assert_retv(id != NULL);
+
+	/* Universal */
+	achievement->validate_type(sd,
+							   success ? ACH_EQUIP_REFINE_SUCCESS : ACH_EQUIP_REFINE_FAILURE,
```

Comment:
Not sure what happened here. If you wanted to align this, please use spaces (else just indent the continuation lines with the standard two tabs, to make better use of the precious horizontal space)